### ISSUE TITLE
Hide the Jupyter Server status bar item when a third party extension kernel is used

### ISSUE THREAD

**Issue by davidanthoff** [microsoft/vscode-jupyter#5156]
Right now, when I select the kernel in the Julia extension (well, on the branch of the Julia extension) as the kernel for a notebook, the status bar item `Jupyter Server: local` is still shown, like here:

![image](https://user-images.githubusercontent.com/1036561/111014433-71878680-8358-11eb-91cc-45b5561cecae.png)

I think ideally that would be completely hidden if the user selected the kernel from the Julia extension.

**Comment by claudiaregio** [comment]
Hi @davidanthoff, could you help me understand how it would be helpful for the Julia community to remove this label/action? This is also the entry point we currently use to allow users to connect to a remote Jupyter server. Can one not connect to a remote Jupyter server and use the Julia kernel for some reason?

**Comment by davidanthoff** [comment]
So my understanding right now is that the structure will be that a notebook can be connected in any of the following ways:
- VS Code kernel implemented in Jupyter extension -> ZMQ -> some kernelspec (could be IJulia.jl)
- VS Code kernel implemented in Jupyter extension -> Jupyter Server protocol -> Local kernelspec (could be IJulia.jl)
- VS Code kernel implemented in Jupyter extension -> Jupyter Server protocol -> Remote kernelspec (could be IJulia.jl)
- VS Code kernel implemented in Julia extension -> Julia kernel implemented in the Julia extension (entirely independent of IJulia.jl and the Jupyter protocol)

So if a user picks the Julia kernel that is implemented in the Julia extension, then I don't think showing options that control how the Jupyter extension might connect to kernels makes sense, because the Jupyter extension is out of the picture in terms of kernels at that point, I think?

Right now we don't plan to support remote kernels natively in the kernel that we are implementing in the Julia extension, but if we decide to change that down the road, then I think the Julia extension would have to show UI that lets a user connect to some remote machine, because that scenario would also bypass any remote support that the Jupyter extension might have.

**Comment by claudiaregio** [comment]
Ah okay I see thank you. Missed the part where it would be hidden if exclusively registered from the Julia extension, would still be available for any user who happens to connect to a Julia kernel through the Jupyter extension.

**Comment by joyceerhl** [comment]
Hi @davidanthoff, this VSIX contains a change which should hide the Jupyter server status bar item https://github.com/microsoft/vscode-jupyter/suites/2424687509/artifacts/51749906 Would love your help verifying that we've resolved the problem for you.

I have tested this change with the .NET Interactive extension's kernels and confirmed that it works. I did also try to verify with the Julia extension by building the notebook branch off the julia-vscode extension source and installing the built VSIX, but I wasn't able to find the Julia extension's kernel provider or notebook creation command (possible I did something wrong).
